### DISCUSSION TITLE
Do a release (will be 3.0)

### DISCUSSION THREAD

**Discussion by cprivitere** [dadoonet/fscrawler#1106] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
No.

The lack of releases for this project are a problem. It's difficult to know who is on what, compare bugs and features with other users, or get a sense if the project is moving forward, how, and with what cadence. It's also not possible to do CI/CD deployment with the current snapshot based releases due to a lack of tagging.

**Describe the solution you'd like**

My suggestion is to cut a release and start moving to semantic versioning again instead of every development being added to a rolling snapshot. 

**Describe alternatives you've considered**

Use the snapshots from the master branch, which are not ideal and lead to version confusion when comparing configurations with other users.

------

Looks like 2.7 got cut ages ago and now we're on 2.10 snapshots. It's basically the same situation but I figured I'd give a last status update. I don't use this at my current company so I will be unsubscribing for the time being.

**Comment by dadoonet** [comment]
I totally agree with you. The thing is that I'd really love to provide [Docker images](https://github.com/dadoonet/fscrawler/pull/820) as part of this release.

Here are the issues/bugs that I'd like to fix before cutting a release:

https://github.com/dadoonet/fscrawler/milestone/24

But I do hear you about that need. I wish that days have more than 24 hours :) 

BTW I'm moving your question to a discussion.

**Comment by afraoucene** [comment]
Hello,

As you know, Elasticsearch 7.16 base of FSCrawler 2.9 reaches end of support since more than one year (6 months for 7.17) as described here https://endoflife.date/elasticsearch,

Can you please do a release for the FSCrwler 2.10 version which is full compatible with Elasticsearch 8.19.x and 9.3.x (now should be 9.4.x ?),

 As already said in old comments, working with snpashot version is a problem and unabling infrastructure upgrade expose us to vulnerabilities and can't be status quo.

Thanks by advance for your aid,

**Comment by dadoonet** [comment, marked as answer]
FYI I'm working on doing a release but before that I will switch the whole `master` branch to 3.0 instead of `2.10`.
The project changed so much since `2.9` that I prefer making it obvious.
